java/sqlcommenter-java/src/main/java/com/google/cloud/sqlcommenter/threadlocalstorage/SpanContextMetadata.java

```diff
@@ -0,0 +1,108 @@
+package com.google.cloud.sqlcommenter.threadlocalstorage;
+
+import javax.annotation.Nullable;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class SpanContextMetadata {
+    private static final Logger logger = Logger.getLogger(SpanContextMetadata.class.getName());
+    private static final String UTF8 = StandardCharsets.UTF_8.toString();
+
+    private final String traceId;
+    private final String spanId;
+    private final byte traceOptions;
+    private final String traceState;
+
+    private SpanContextMetadata(String traceId, String spanId, byte traceOptions, @Nullable String traceState) {
+        this.traceId = traceId;
+        this.spanId = spanId;
+        this.traceOptions = traceOptions;
+        this.traceState = traceState;
+    }
+
+    public static SpanContextMetadata fromOpenCensusContext(io.opencensus.trace.SpanContext spanContext) {
+        if (spanContext == null || !spanContext.isValid()) {
+            return null;
+        }
+        String traceId = spanContext.getTraceId().toLowerBase16();
+        String spanId = spanContext.getSpanId().toLowerBase16();
+        byte traceOptions = spanContext.getTraceOptions().getByte();
+        String traceStateStr = null;
+
+        io.opencensus.trace.Tracestate traceState = spanContext.getTracestate();
+        if (!traceState.getEntries().isEmpty()) {
```

Comment:
Nit: this code could be less nested if we inverted this condition and returned early so on isEmpty() immediately return, otherwise build traceStateStr.